core/src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -106,7 +106,7 @@ public boolean isDeprecatedSetting(String setting) {
         token = parser.nextToken();
         if (token == XContentParser.Token.END_OBJECT) {
             // empty query
-            return EmptyQueryBuilder.PROTOTYPE;
+            return new EmptyQueryBuilder();
```

Comment:
so now that we rewrite we potentially return an EmptyQueryBuilder form a rewrite method. for instance with `new WrapperQueryBuilder("{}")` this is problematic if the `WrapperQueryBuilder` had a query name which is inherited but ignored by the old version to make all assumptions hold we have to upgrade this query builder to a proper one., makes sense?